src/herbivoresAndCarnivores.js

```diff
@@ -1,17 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    this.alive = [];
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100, hidden = false) {
+    super(name, health);
+
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    const isHerbivore = animal instanceof Herbivore;
+
+    if (isHerbivore && !animal.hidden) {
+      animal.health -= 50;
+
+      if (animal.health <= 0) {
```

Comment:
it won't get less than 0, just 0
```suggestion
      if (animal.health === 0) {
```
or 
```suggestion
      if (!animal.health) {
```